utils/preferences/tests/dtf/mod.rs

```diff
@@ -0,0 +1,84 @@
+mod data_provider;
+mod options;
+mod preferences;
+
+use data_provider::get_defaults;
+use icu_datetime::options::length;
+use icu_datetime::options::preferences::HourCycle;
+use icu_locid::{extensions_unicode_key, LanguageIdentifier, Locale};
+use icu_preferences::{preferences, Preferences};
+use options::{DayPeriod, LocaleMatcher};
+use preferences::{Calendar, NumberingSystem};
+
+preferences!(
+    DTFPreferencesBag,
+    DTFDefaultPreferencesBag,
+    DTFResolvedPreferencesBag,
+    {
+        hour_cycle => Option<HourCycle>, HourCycle, Some(extensions_unicode_key!("hc")),
+        calendar => Option<Calendar>, Calendar, Some(extensions_unicode_key!("ca")),
+        numbering_system => Option<NumberingSystem>, NumberingSystem, Some(extensions_unicode_key!("nu"))
+    },
+    DTFOptionsBag,
+    {
+        date_length => Option<length::Date>,
+        time_length => Option<length::Time>,
+        day_period => Option<DayPeriod>,
+        locale_matcher => Option<LocaleMatcher>,
+        time_zone => Option<bool>
```

Comment:
Note: Time Zone is an example of a property that could come from the options bag but could also come from the locale/preferences.

Reply:
I'm questioning that paradigm. I think that we should not take Locale tz into account when formatting a date, and I believe it is unnecessary to use it as options. Instead, I think developer should resolve timezone of each date before passing that date to DTF. DTF should not convert timezones.